proposer/proposer.go

```diff
@@ -306,6 +309,20 @@ func (p *Proposer) ProposeTxList(
 	return nil
 }
 
+// updateProposingTicker updates the internal proposing timer.
+func (p *Proposer) updateProposingTicker() {
+	var duration time.Duration
+	if p.proposingInterval != nil {
+		duration = *p.proposingInterval
+	} else {
+		rand.Seed(time.Now().UnixNano())
+		// Random number between 12 - 60
+		randomSeconds := rand.Intn((60 - 11)) + 12
+		duration = time.Duration(randomSeconds) * time.Second
+	}
+	p.proposingTimer = time.NewTimer(duration)
```

Comment:
need stop previous timer (if call `updateProposingTicker` multiple times

Reply:
changed: https://github.com/taikoxyz/taiko-client/pull/76/commits/ca4253ecfbb6ffa33f43611d39b5d2e4c7b2221a

Reply:
I see that it calls `updateProposingTicker` after `case <-p.proposingTimer.C` in each loop
maybe no need stops previous timer (be careful when we have move `select case` without returns

Reply:
> I see that it calls `updateProposingTicker` after `case <-p.proposingTimer.C` in each loop maybe no need stops previous timer (be careful when we have move `select case` without returns

i added it for safety.